vault/client.go

```diff
@@ -89,9 +89,9 @@ func authenticate(c *vaultapi.Client, authType string, params map[string]string)
 			"password": password,
 		})
 	case "kubernetes":
-		jwt, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
-		if err != nil {
-			return err
+		jwt, readErr := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+		if readErr != nil {
+			return readErr
```

Comment:
This looks odd, but it's because just `err` overrides the named return `err` argument, causing the `err` on line 96 to go unused, or at least staticcheck thinks so 👀